### DISCUSSION TITLE
Automatic Voltage Select

### DISCUSSION THREAD

**Discussion by Ancyker** [sanni/cartreader#625] [Ideas & Suggestions]
Been working on this a bit, but almost ready. Just need to finish designing the PCB now.

https://youtube.com/shorts/Uz5B7tNjIh4

With this modification, voltage is selected by the software/code.

**Comment by Ancyker** [comment]
I've moved it to a perfboard, looks a lot simpler now...

![](https://cdn.discordapp.com/attachments/1038272253852401694/1040839652116004964/2022-11-11_23.00.41.jpg)
![](https://cdn.discordapp.com/attachments/1038272253852401694/1040839323353890836/2022-11-11_23.01.13.jpg)

It does replace the power LED though.

**Comment by sanni** [comment]
Great work. 👍 
Even seems small enough to be used as a drop-in replacement for the switch without the need to remake the main PCB.

**Comment by Ancyker** [comment]
PR #730 will put support for this on the main PCB.

I wonder if we should look into a 3-position switch so that it can be toggled 5V/3.3V/Auto? Alternatively, we could add a menu option to force a voltage, but I think we'd need some feedback for that somewhere to let you know what it's set to. I wish we had more pins, lol.

I'd hate to add another component just for vselect though; @sanni, is there anything you were wanting to do that needed a couple more pins? I was playing around [with this](https://ww1.microchip.com/downloads/en/devicedoc/20001952c.pdf), but it seems overkill just for a couple of status LEDs:
![2022-12-03 22 14 55](https://user-images.githubusercontent.com/7105333/224102118-1c67aca2-6489-4c8f-b0fb-1ba78e15029a.jpg)
